compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -223,6 +235,36 @@ def task_group_id(self):
     def task_group_id(self, task_group_id: UUID_LIKE_T | None):
         self._task_group_id = as_optional_uuid(task_group_id)
 
+    @property
+    def user_endpoint_config(self) -> dict[str, t.Any] | None:
+        """
+        The endpoint configuration values, as described and allowed by endpoint
+        administrators, that this instance is currently associated with.
+
+        Must be a dict or None. Set by simple assignment::
+
+            >>> from globus_compute_sdk import Executor
+            >>> uep_config = {"foo": "bar"}
+            >>> gce = Executor(user_endpoint_config=uep_config
+
+            # May also alter after construction:
+            >>> gce.user_endpoint_config = uep_config
+        """
+        return self._user_endpoint_config
+
+    @user_endpoint_config.setter
+    def user_endpoint_config(self, val: dict | None):
+        if val is not None:
+            if not isinstance(val, dict):
+                raise TypeError(
+                    f"Config must be of type dict, not {type(val).__name__}"
+                )
+            try:
+                json.dumps(val)
+            except TypeError as err:
+                raise TypeError(f"Invalid config: {str(err)}")
+        self._user_endpoint_config = val
```

Comment:
Nice validation; it's too bad we don't have access to Pydantic ... ;-)

Was going to suggest testing JSON serialization, then _actually read the code_.  :boom: 

One suggestion: give the higher-level error in the TypeError exception.  Perhaps something like "Configuration must be JSON-serializable."  Give folks something more to Google when it explodes for them.

Reply:
Nice suggestion; I'll implement in a fresh commit.